integration-tests/test_connection.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+
+# the tests need a valid connection to insights so therefore the subman registration
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_connection(insights_client):
+    """
+    Test insights-client --test-connection
+    `test_connection` method returns 0 if success, 1 if failure
+    """
+    url_test = "End Upload URL Connection Test: SUCCESS"
+    api_test = "End API URL Connection Test: SUCCESS"
+
+    test_connection = insights_client.run("--test-connection")
+    assert url_test in test_connection.stdout
+    assert api_test in test_connection.stdout
+
+
+def test_http_timeout(insights_client):
+    """
+    Test http_timeout config option
+    Set http_timeout to a very low value, run --test-connection and check
+    the set time
+    """
+    insights_client.config.http_timeout = 0.1
+    insights_client.config.save()
+
+    output = insights_client.run("--test-connection", check=False)
+    assert output.returncode == 1
+    assert "read timeout=0.1" in output.stdout
+
+
+def test_noauth_proxy_connection(insights_client, test_config):
+    """
+    Test no-auth proxy option with --test-connection option
+    """
+    url_test = "End Upload URL Connection Test: SUCCESS"
+    api_test = "End API URL Connection Test: SUCCESS"
+
+    no_auth_proxy: str = (
+        "http://"
+        + test_config.get("noauth_proxy", "host")
+        + ":"
+        + str(test_config.get("noauth_proxy", "port"))
+    )
+    insights_client.config.proxy = no_auth_proxy
+    insights_client.config.save()
+
+    test_connection = insights_client.run("--test-connection")
+    assert url_test in test_connection.stdout
+    assert api_test in test_connection.stdout
+
+
+def test_auth_proxy_connection(insights_client, test_config):
+    """
+    Test auth proxy option with --test-connection
+    """
+    url_test = "End Upload URL Connection Test: SUCCESS"
+    api_test = "End API URL Connection Test: SUCCESS"
+
+    auth_proxy: str = (
+        "http://"
+        + test_config.get("auth_proxy", "username")
+        + ":"
+        + test_config.get("auth_proxy", "password")
+        + "@"
+        + test_config.get("auth_proxy", "host")
+        + ":"
+        + str(test_config.get("auth_proxy", "port"))
+    )
+    insights_client.config.proxy = auth_proxy
+    insights_client.config.save()
+    test_connection = insights_client.run("--test-connection")
+    assert url_test in test_connection.stdout
+    assert api_test in test_connection.stdout
+
+
+def test_wrong_url_connection(insights_client):
+    """
+    Test wrong url option with --test-connection
+    """
+    wrong_url = (
+        "Max retries exceeded with url: //no-such-insights-url.example.com/something"
```

Comment:
in output lines with this text I do not see the hostname being attempted, only the paths; for example:
```
HTTPSConnectionPool(host='no-such-insights-url.example.com', port=443): Max retries exceeded with url: /something/uploads/ (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f751c80f1c0>: Failed to establish a new connection: [Errno -2] Name or service not known'))
HTTPSConnectionPool(host='no-such-insights-url.example.com', port=443): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f751c3e0190>: Failed to establish a new connection: [Errno -2] Name or service not known'))
HTTPSConnectionPool(host='no-such-insights-url.example.com', port=443): Max retries exceeded with url: /r (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f751c80f760>: Failed to establish a new connection: [Errno -2] Name or service not known'))
HTTPSConnectionPool(host='no-such-insights-url.example.com', port=443): Max retries exceeded with url: /r/insights (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f751c3e0ac0>: Failed to establish a new connection: [Errno -2] Name or service not known'))
HTTPSConnectionPool(host='no-such-insights-url.example.com', port=443): Max retries exceeded with url: /r/insights (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f751c3e0ac0>: Failed to establish a new connection: [Errno -2] Name or service not known'))
```
also, "Max retries exceeded with url" is a message printed by `requests`, and thus may change depending on its version; I'd rather check for strings printed by insights-core itself (which can be kept static) rather than relying on what a 3rd party module prints as default exception representation